android/player/src/main/java/com/intuit/playerui/android/asset/RenderableAsset.kt

```diff
@@ -109,34 +111,36 @@ public constructor(public val assetContext: AssetContext) : NodeWrapper {
      * also automatically caches the instance of the [View] and detects
      * when it needs to reconstruct or rehydrate.
      */
-    private fun render(): View = cachedAssetView.let { (cachedAssetContext, cachedView) ->
-        requireContext()
-        when {
-            // View not found. Create and hydrate.
-            cachedView == null -> {
-                renewHydrationScope("recreating view")
-                initView().also { it.hydrate() }
-            } // View found, but contexts are out of sync. Remove cached view and create and hydrate.
-            cachedAssetContext?.context != context || cachedAssetContext?.asset?.type != asset.type -> {
-                renewHydrationScope("recreating view")
-                cachedView.removeSelf()
-                initView().also { it.hydrate() }
-            }
-            // View found, but assets are out of sync. Rehydrate. It is possible for the hydrate
-            // implementation to throw [StaleViewException] to signify that the view is out of sync.
-            // This can only be done from invalidateView, so we have a guarantee that the view has
-            // already been removed from the cache.
-            !cachedAssetContext.asset.nativeReferenceEquals(asset) ->
-                try {
-                    cachedView.also(::rehydrate)
-                } catch (exception: StaleViewException) {
-                    player.logger.info("re-rendering due to stale child: ${exception.assetContext.id}")
-                    render()
+    private suspend fun render(): View = withContext(Dispatchers.Default) {
```

Comment:
I don't think `render` should change, or even be aware of what coroutine context it's executing on.

Reply:
this would lead to a problem of this being called on `main` in nested [hydrates](https://github.com/player-ui/player/blob/main/android/player/src/main/java/com/intuit/playerui/android/asset/SuspendableAsset.kt#L61)